File: module-8-harness/solutions/main.ts
/**
 * Module 8: Harness CLI
 * Usage: npx tsx module-8-harness/main.ts "task" [workspace] [--resume session-id]
 */
import "dotenv/config";
import { runHarness } from "./harness.js";
import { listSessions } from "./session.js";

async function main() {
  const args = process.argv.slice(2);

  // Handle --list flag
  if (args[0] === "--list") {
    const sessions = await listSessions();
    if (sessions.length === 0) {
      console.log("No sessions found.");
    } else {
      console.log("Sessions:");
      for (const s of sessions) {
        console.log(`  ${s.id} — ${s.task.slice(0, 60)} (${s.updatedAt})`);
      }
    }
    return;
  }

  // Handle --resume flag
  const resumeIdx = args.indexOf("--resume");
  let sessionId: string | undefined;
  if (resumeIdx !== -1) {
    sessionId = args[resumeIdx + 1];
    args.splice(resumeIdx, 2);
  }

  const task = args[0];
  if (!task && !sessionId) {
    console.error("Usage: npx tsx module-8-harness/main.ts \"<task>\" [workspace] [--resume session-id]");
    console.error("       npx tsx module-8-harness/main.ts --list");
    process.exit(1);
  }

  const workspace = args[1] || process.cwd();
  const result = await runHarness(task || "(resumed)", workspace, sessionId);
  console.log(`\n=== Result ===\n${result}`);
}

main().catch(console.error);

File: module-8-harness/solutions/python/main.py
"""Module 8: Harness CLI."""
import sys
import os

sys.path.insert(0, os.path.dirname(__file__))

from dotenv import load_dotenv
load_dotenv()

from harness import run_harness
from session import list_sessions


def main():
    args = sys.argv[1:]

    if args and args[0] == "--list":
        sessions = list_sessions()
        if not sessions:
            print("No sessions found.")
        else:
            print("Sessions:")
            for s in sessions:
                print(f"  {s['id']} — {s['task'][:60]} ({s['updatedAt']})")
        return

    session_id = None
    if "--resume" in args:
        idx = args.index("--resume")
        session_id = args[idx + 1]
        args = args[:idx] + args[idx + 2:]

    task = args[0] if args else None
    if not task and not session_id:
        print('Usage: python main.py "<task>" [workspace] [--resume session-id]')
        print("       python main.py --list")
        sys.exit(1)

    workspace = args[1] if len(args) > 1 else os.getcwd()
    result = run_harness(task or "(resumed)", workspace, session_id)
    print(f"\n=== Result ===\n{result}")


if __name__ == "__main__":
    main()
